datafusion/common/src/scalar/mod.rs

```diff
@@ -1063,7 +1063,6 @@ impl ScalarValue {
 
     /// Create an one value in the given type.
     pub fn new_one(datatype: &DataType) -> Result<ScalarValue> {
-        assert!(datatype.is_primitive());
```

Comment:
removing the assert (as any unhandled type will cause a runtime error) makes a lot of sense to me.